packages/opencensus-instrumentation-ioredis/test/test-ioredis.ts

```diff
@@ -0,0 +1,202 @@
+import * as assert from 'assert';
+import * as ioredis from 'ioredis';
+
+import {plugin} from '../src';
+
+/** Collects ended root spans to allow for later analysis. */
+class RootSpanVerifier implements SpanEventListener {
+  endedRootSpans: RootSpan[] = [];
+
+  onStartSpan(span: RootSpan): void {
+    return;
+  }
+  onEndSpan(root: RootSpan) {
+    this.endedRootSpans.push(root);
+  }
+}
+
+/**
+ * Asserts root spans attributes.
+ * @param rootSpanVerifier An instance of rootSpanVerifier to analyse RootSpan
+ * instances from.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpan(
+    rootSpanVerifier: RootSpanVerifier, expectedName: string,
+    expectedKind: SpanKind, verifyAttribute?: (span: Span) => boolean) {
+  assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+  assert.strictEqual(rootSpanVerifier.endedRootSpans[0].spans.length, 1);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].name, expectedName);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].kind, expectedKind);
+  if (typeof verifyAttribute === 'function') {
+    for (const span of rootSpanVerifier.endedRootSpans[0].spans) {
+      assert(verifyAttribute(span), 'failed to verify attribute');
+    }
+  }
+}
+
+describe('IORedisPlugin', () => {
+  // For these tests, mongo must be runing. Add OPENCENSUS_REDIS_TESTS to run
+  // these tests.
+  const OPENCENSUS_REDIS_TESTS = process.env.OPENCENSUS_REDIS_TESTS;
+  const OPENCENSUS_REDIS_HOST = process.env.OPENCENSUS_REDIS_HOST;
+  const shouldTest = OPENCENSUS_REDIS_TESTS === '1';
+  if (!shouldTest) {
+    console.log('Skipping test-redis. Run REDIS to test');
+  }
+
+  const URL = `redis://${OPENCENSUS_REDIS_HOST || 'localhost'}:6379`;
+  const VERSION = '4.6.2';
+
+  const tracer = new CoreTracer();
+  const rootSpanVerifier = new RootSpanVerifier();
+  let client: ioredis.Redis;
+
+  before((done) => {
+    tracer.start({samplingRate: 1, logger: logger.logger(1)});
+    tracer.registerSpanEventListener(rootSpanVerifier);
+    plugin.enable(ioredis, tracer, VERSION, {}, '');
+    client = new ioredis(URL);
+    client.on('error', (err: Error) => {
+      done(err);
+    });
+    client.on('ready', done);
+  });
+
+  beforeEach(async function redisBeforeEach() {
+    // Skiping all tests in beforeEach() is a workarround. Mocha does not work
+    // properly when skiping tests in before() on nested describe() calls.
+    // https://github.com/mochajs/mocha/issues/2819
+    if (!shouldTest) {
+      this.skip();
+    }
+    rootSpanVerifier.endedRootSpans = [];
+    // Non traced insertion of basic data to perform tests
+    await client.set('test', 'data');
+  });
+
+  afterEach(async () => {
+    await client.del('hash');
+  });
+
+  after(() => {
+    if (client) {
+      client.quit();
+    }
+  });
+
+  /** Should intercept query */
```

Comment:
Optional: can this comment be removed since it's implied by the `describe` and `it` messages below? If not, could they be rewritten in a such a way that it could be removed?